src/cgame/cg_tutorial.cpp

```diff
@@ -444,13 +444,32 @@ static void CG_HumanText( char *text, playerState_t *ps )
 		}
 	}
 
-	if ( upgrade == UP_NONE ||
-	     ( upgrade > UP_NONE && BG_Upgrade( upgrade )->usable ) )
+	// Find next weapon in inventory.
+	// FIXME: factorize code with CG_NextWeapon_f() from src/cgame/cg_weapons.cpp
+	weapon_t nextWeapon = WP_NONE;
+	for ( weapon_t w = WP_NONE; w < WP_NUM_WEAPONS; w = weapon_t( w + 1 ) )
```

Comment:
In this case since we do need to know the weapon, a simpler algorithm would be to find the first weapon that is not the current wepano.

Reply:
That's what the code does, but it also takes care of the case of the current weapon is the last one of the list.

Reply:
- if current weapon: continue
- if first weapon before finding current weapon: remember
- if first weapon after finding current weapon: remember, stop

Reply:
This code now reuses the function from `cg_weapons.cpp`.